ui/perfherder/helpers.js

```diff
@@ -0,0 +1,334 @@
+};
+
+export const getStdDev = function getStandardDeviation(values, avg) {
+  if (values.length < 2) {
+    return undefined;
+  }
+
+  if (!avg) avg = calcAverage(values);
+
+  return Math.sqrt(
+    values.map(v => (v - avg) ** 2).reduce(function (a, b) { return a + b; }) / (values.length - 1));
+};
+
+// If a set has only one value, assume average-ish-plus standard deviation, which
+// will manifest as smaller t-value the less items there are at the group
+// (so quite small for 1 value). This default value is a parameter.
+// C/T mean control/test group (in our case original/new data).
+export const getTTest = function getTTest(valuesC, valuesT, stddev_default_factor) {
+  const lenC = valuesC.length;
+  const lenT = valuesT.length;
+
+  if (!lenC || !lenT) {
+    return 0;
+  }
+
+  const avgC = calcAverage(valuesC);
+  const avgT = calcAverage(valuesT);
+  let stddevC = (lenC > 1 ? getStdDev(valuesC, avgC) : stddev_default_factor * avgC);
+  let stddevT = (lenT > 1 ? getStdDev(valuesT, avgT) : stddev_default_factor * avgT);
+
+  if (lenC === 1) {
+    stddevC = valuesC[0] * stddevT / avgT;
+  } else if (lenT === 1) {
+    stddevT = valuesT[0] * stddevC / avgC;
+  }
+
+  const delta = avgT - avgC;
+  const stdDiffErr = (
+    Math.sqrt(
+      (stddevC * stddevC / lenC) // control-variance / control-size
+        +
+      (stddevT * stddevT / lenT),
+    )
+  );
+
+  return delta / stdDiffErr;
+};
+// TODO many of these are only used in one controller so can likely be moved
+// into the appropriate react component
+
+// eslint-disable-next-line arrow-body-style
+const numericCompare = (a, b) => { return a < b ? -1 : (a > b ? 1 : 0); };
+
+const analyzeSet = (values, testName) => {
+  let average;
+  let stddev = 1;
+
+  if (testName === 'Noise Metric') {
+      average = Math.sqrt(values.map(x => x ** 2).reduce((a, b) => a + b, 0));
+  } else {
+      average = calcAverage(values);
+      stddev = getStdDev(values, average);
+  }
+
+  return {
+      average,
+      stddev,
+      stddevPct: Math.round(calcPercentOf(stddev, average) * 100) / 100,
+      // TODO verify this is needed
+      // We use slice to keep the original values at their original order
+      // in case the order is important elsewhere.
+      runs: values.slice().sort(numericCompare),
+  };
+};
+
+const getClassName = (newIsBetter, oldVal, newVal, absTValue) => {
+  // NOTE: we care about general ratio rather than how much is new compared
+  // to old - this could end up with slightly higher or lower threshold
+  // in practice than indicated by DIFF_CARE_MIN. E.g.:
+  // - If old is 10 and new is 5, then new = old -50%
+  // - If old is 5 and new is 10, then new = old + 100%
+  // And if the threshold was 75% then one would matter and the other wouldn't.
+  // Instead, we treat both cases as 2.0 (general ratio), and both would matter
+  // if our threshold was 75% (i.e. DIFF_CARE_MIN = 1.75).
+  if (!oldVal || !newVal) {
```

Comment:
@jmaher This comment looks like its referring to a variable that doesn't exist (DIFF_CARE_MIN). Can the reference to it be removed or else rewritten a bit more succinctly? :)

Reply:
I looked back in history and see no clues.  We copied this algorithm from a repo as close to verbatim as we could- it was a python repo that we had and ran the alerts from previously.  I suspect the comment is from there and removing references to DIFF_CARE_MIN seems like a good idea here.  The question is what could we write this to be- maybe @klahnakoski could help define a better comment?  My attempt would be:

This code looks for an relative change in the absolute value and will produce either:
* no change
* maybe a change
* improvement
* regression

Reply:
I replaced it with 'Returns a class name, if any, based on a relative change in the absolute value' since the class names are fairly self-explanatory (per your bullet points).